File: BBGestureBack_Full_oc/Classes/Home/Index/HomeViewController.m
//  希望您的举手之劳，能为我点颗赞，谢谢~
//  代码地址: https://github.com/Bonway/BBGestureBack
//  BBGestureBack
//  Created by Bonway on 2016/3/17.
//  Copyright © 2016年 Bonway. All rights reserved.
//


#import "HomeViewController.h"
#import "ViewController.h"
@interface HomeViewController ()

@end

@implementation HomeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    UIButton *sysBtn = [[UIButton alloc]initWithFrame:CGRectMake(100, 100, 100, 100)];
    sysBtn.backgroundColor = [UIColor blueColor];
    [sysBtn setTitle:@"sysPush" forState:UIControlStateNormal];
    [sysBtn addTarget:self action:@selector(sysPushClick) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:sysBtn];
    
}

-(void)sysPushClick{
    [self.navigationController pushViewController:[ViewController new] animated:YES];
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}



@end

File: BBGestureBack_SysFull_swift/Classes/Home/Index/HomeViewController.swift
//
//  HomeViewController.swift
//  BBGestureBack_SysFull_swift
//
//  Created by Bonway on 2018/5/29.
//  Copyright © 2018年 Bonway. All rights reserved.
//

import UIKit

class HomeViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()

        //iOS12.1 UINavigationController + UITabBarController（ UITabBar 磨砂），在 系统的 popViewControllerAnimated 会遇到tabbar布局错乱的问题，所以需添加以下方法。
        //当然你也可以自己选择添加到底层的navigationController中，进行处理。
        UITabBar.appearance().isTranslucent = false
        //---------------------------------------
        
        view.backgroundColor = UIColor.white
        let sysBtn = UIButton()
        sysBtn.backgroundColor = UIColor.blue
        sysBtn.frame = CGRect(x: 100, y: 100, width: 110, height: 100)
        sysBtn.setTitle("sysPush", for: UIControlState.normal)
        sysBtn.addTarget(self, action: #selector(sysPushClick), for: UIControlEvents.touchUpInside)
        
        self.view.addSubview(sysBtn)
    }

    
    @objc func sysPushClick() {
        self.navigationController?.pushViewController(ViewController(), animated: true)
    }
    
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    

}
